src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -77,15 +95,14 @@ export class VSCodeNotebookController implements Disposable {
         private readonly kernelProvider: IKernelProvider,
         private readonly preferredRemoteKernelIdProvider: PreferredRemoteKernelIdProvider,
         private readonly context: IExtensionContext,
-        private readonly notebookControllerManager: INotebookControllerManager,
         private readonly pathUtils: IPathUtils,
         disposableRegistry: IDisposableRegistry,
         private readonly languageService: NotebookCellLanguageService,
         private readonly workspace: IWorkspaceService,
-        private readonly setAsActiveControllerForTests: (
-            controller: VSCodeNotebookController,
-            notebook: NotebookDocument
-        ) => Promise<void>
+        private readonly localOrRemoteKernel: 'local' | 'remote',
```

Comment:
This controller only needs to know whether its a local or remove controller.
This way, if tomorrow we support both together, we don't need to make too many changes.
I.e. don't assume that we can only even have one type at a time...